OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -40,24 +40,37 @@ public class McvManagerBotModuleInfo : ConditionalTraitInfo
 		public readonly int MinBaseRadius = 2;
 
 		[Desc("Maximum distance in cells from center of the base when checking for MCV deployment location.",
-			"Only applies if RestrictMCVDeploymentFallbackToBase is enabled and there's at least one construction yard.")]
+			"AI will not send MCVs to locations containing enemies within the MaxBaseRadius of that location.",
+			"AI will not send MCVs to locations containing its own MCVs or construction yards within the MaxBaseRadius of that location.",
+			"In the case of less than 1 construction yard, the AI will choose a random new base center and place the MCV within the MaxBaseRadius of that location.",
+			"In the case of at least 1 construction yard, the AI will choose a location nearer resource fields outside of all bases MaxBaseRadius up to world.Map.Grid.MaximumTileSearchRange.")]
 		public readonly int MaxBaseRadius = 20;
 
-		[Desc("Should deployment of additional MCVs be restricted to MaxBaseRadius if explicit deploy locations are missing or occupied?")]
-		public readonly bool RestrictMCVDeploymentFallbackToBase = true;
```

Comment:
We should not forget to add an update rule for this. (You can try adding one yourself, but it might be hard for someone not familiar with the update rule system.)

Reply:
Can you advise me whereabouts I would place this update rule? I would like to have a crack at it but can watch someone else if I'm unable to do it :).

Reply:
This is where the problem starts: Until #17698 is merged there is no place for new update rules. You can however already start writing the class and then later move it to the correct place. I would advise you to look at the [existing update rules](https://github.com/OpenRA/OpenRA/tree/bleed/OpenRA.Mods.Common/UpdateRules/Rules) as examples on how to write one.

Reply:
Fwiw, since #17698 was merged quickly, you can directly add the rule to the release-20200202 directory and to https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/UpdateRules/UpdatePath.cs#L64-L78 when working on it.

Reply:
I rebased my branch onto bleed and added the update rule. Please let me know if anything was done incorrectly.

Reply:
This doesn't look correct (it now wants to merge 26 commits). Can you please rebase against latest bleed again and then force push? (Let me know if there is any trouble; it looks to me like you'll have to manually resolve conflicts, one of the reasons I wouldn't recommend using `git merge`.)
The update rule does look good on a first glance however.

Reply:
Yeah I don't think that worked. I hadn't used git merge, I originally used rebase so I'm not sure why it's not worked. Probably user error.

I can reopen this PR with the commits cherry-picked if that's easier ?

Reply:
Or maybe it did work and I spoke too soon :)

Reply:
Yes, looks like the rebase worked. Well done. :)